apps/api/src/server/applications/subscriptions/__tests__/subscriptions.test.js

```diff
@@ -509,4 +509,95 @@ describe('subscriptions', () => {
 			});
 		}
 	});
+
+	describe('prepareInput', () => {
+		/**
+		 * @typedef {Object} Test
+		 * @property {string} name
+		 * @property {*} request
+		 * @property {import('@prisma/client').Prisma.SubscriptionCreateInput} want
+		 */
+		/** @type {Test[]} */
+		const tests = [
+			{
+				name: 'should prepare basic request, setting all subscription fields false',
+				request: {
+					caseReference: 'abc',
+					emailAddress: 'hello@example.com',
+					subscriptionTypes: []
+				},
+				want: {
+					caseReference: 'abc',
+					emailAddress: 'hello@example.com',
+					subscribedToAllUpdates: false,
+					subscribedToApplicationDecided: false,
+					subscribedToApplicationSubmitted: false,
+					subscribedToRegistrationOpen: false,
+					startDate: null,
+					endDate: null
+				}
+			},
+			{
+				name: 'should set only all updates subscription fields true',
+				request: {
+					caseReference: 'abc',
+					emailAddress: 'hello@example.com',
+					subscriptionTypes: ['allUpdates', 'applicationDecided']
+				},
+				want: {
+					caseReference: 'abc',
+					emailAddress: 'hello@example.com',
+					subscribedToAllUpdates: true,
+					subscribedToApplicationDecided: false,
+					subscribedToApplicationSubmitted: false,
+					subscribedToRegistrationOpen: false,
+					startDate: null,
+					endDate: null
+				}
+			},
+			{
+				name: 'should set appropriate subscription fields true',
+				request: {
+					caseReference: 'abc',
+					emailAddress: 'hello@example.com',
+					subscriptionTypes: ['applicationDecided', 'registrationOpen']
+				},
+				want: {
+					caseReference: 'abc',
+					emailAddress: 'hello@example.com',
+					subscribedToAllUpdates: false,
+					subscribedToApplicationDecided: true,
+					subscribedToApplicationSubmitted: false,
+					subscribedToRegistrationOpen: true,
+					startDate: null,
+					endDate: null
+				}
+			},
+			{
+				name: 'should set start and end as dates',
+				request: {
+					caseReference: 'abc',
+					emailAddress: 'hello@example.com',
+					subscriptionTypes: ['applicationDecided', 'registrationOpen'],
+					startDate: '2023-07-03T07:44:00Z',
+					endDate: '2023-07-15T07:44:00Z'
+				},
+				want: {
+					caseReference: 'abc',
+					emailAddress: 'hello@example.com',
+					subscribedToAllUpdates: false,
+					subscribedToApplicationDecided: true,
+					subscribedToApplicationSubmitted: false,
+					subscribedToRegistrationOpen: true,
+					startDate: new Date('2023-07-03T07:44:00Z'),
+					endDate: new Date('2023-07-15T07:44:00Z')
+				}
+			}
+		];
+		for (const { name, request, want } of tests) {
```

Comment:
There is a jest construct for this kind of thing, it.each. I've used it on the doc publishing function

Reply:
Ah yes I did notice that, but promptly forgot when I wrote my tests. I've refactored all the tests in this file:
https://github.com/Planning-Inspectorate/back-office/pull/1120/commits/5efd2db18dfcb7e000ac138255622dbb43adbeb5